repos/system_upgrade/common/actors/addupgradebootentry/libraries/addupgradebootentry.py

```diff
@@ -33,6 +33,26 @@ def add_boot_entry(configs=None):
             # otherwise the new boot entry will not be set as default
             # See https://bugzilla.redhat.com/show_bug.cgi?id=1764306
             run(['/usr/sbin/zipl'])
+
+        if debug:
+            source_kernel_cmdline_msgs = api.consume(KernelCmdline)
+            source_kernel_cmdline = next(source_kernel_cmdline_msgs, None)
+            if not source_kernel_cmdline:
+                # Highly unlikely - something is wrong.
+                raise StopActorExecutionError('Actor did not receive any KernelCmdline messages.')
+            if next(source_kernel_cmdline_msgs, None):
+                api.current_logger().warning('Received more than one KernelCmdline message.')
+
+            has_source_debug_kernelopt = any(
+                (kernelopt.key == 'debug' for kernelopt in source_kernel_cmdline.parameters)
+            )
+
+            if not has_source_debug_kernelopt:
```

Comment:
@MichalHe  `KernelCmdline` contain just kernel args that have been set for the current system run, from `/proc/cmdline`. Someone could add the `debug` option during the boot and in such case I do not think it's expected in such a case preserve the option. For that, we would need to parse option for the current boot entry, not frmo the current boot. Which I do not think it's reasonable to do now. I think we should go with the previous solution and comment. wdyt?